fox.go

```diff
@@ -222,29 +226,31 @@ func DefaultMethodNotAllowedHandler() HandlerFunc {
 	}
 }
 
-func defaultRedirectTrailingSlash() HandlerFunc {
-	return func(c Context) {
-		req := c.Request()
+func defaultRedirectTrailingSlash(c Context) {
+	req := c.Request()
 
-		code := http.StatusMovedPermanently
-		if req.Method != http.MethodGet {
-			// Will be redirected only with the same method (SEO friendly)
-			code = http.StatusPermanentRedirect
-		}
+	code := http.StatusMovedPermanently
+	if req.Method != http.MethodGet {
+		// Will be redirected only with the same method (SEO friendly)
+		code = http.StatusPermanentRedirect
+	}
 
-		var url string
-		if len(req.URL.RawPath) > 0 {
-			url = fixTrailingSlash(req.URL.RawPath)
-		} else {
-			url = fixTrailingSlash(req.URL.Path)
-		}
+	var url string
+	if len(req.URL.RawPath) > 0 {
+		url = fixTrailingSlash(req.URL.RawPath)
+	} else {
+		url = fixTrailingSlash(req.URL.Path)
+	}
 
-		if url[len(url)-1] == '/' {
-			localRedirect(c.Writer(), req, path.Base(url)+"/", code)
-			return
-		}
-		localRedirect(c.Writer(), req, "../"+path.Base(url), code)
+	if url[len(url)-1] == '/' {
```

Comment:
Un `index out of range` potentiel  si l'URL est vide. Est-ce que ça pourrait arrivé?